bot/exts/evergreen/issues.py

```diff
@@ -26,44 +28,59 @@ class Issues(commands.Cog):
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
+        self.repos = []
+        self.get_pydis_repos.start()
+
+    @tasks.loop(minutes=30)
+    async def get_pydis_repos(self) -> None:
+        """Get all python-discord repositories on github."""
+        async with self.bot.http_session.get(PYDIS_REPOS) as resp:
+            if resp.status == 200:
+                data = await resp.json()
+                for repo in data:
+                    self.repos.append(repo["full_name"].split("/")[1])
+            else:
+                log.debug(f"Failed to get latest Pydis repositories. Status code {resp.status}")
+
+    @staticmethod
+    def check_in_block(message: discord.Message, repo_issue: str) -> bool:
+        """Check whether the <repo>#<issue> is in codeblocks."""
+        block = (
+            re.findall(r"```([\s\S]*)?```", message.content)
+            or re.findall(r"```*\n([\s\S]*)?\n```", message.content)
+            or re.findall(r"```*([\s\S]*)?\n```", message.content)
+            or re.findall(r"```*\n([\s\S]*)?```", message.content)
+            or re.findall(r"`([\s\S]*)?`", message.content)
+        )
```

Comment:
I think at this point all these regex strings should be compiled as constants and put into a tuple so this can be iterated over easily instead of having many `or` statements.

Reply:
Also, what are these regexes for? Couldn’t just just one smarter regex?

Reply:
Well ig the doc string is self-explanatory, they are for checking if the `<repo>#<issue>` part of a message are in code-blocks or not.
Couldn't find a smarter way 😅

Reply:
What's the difference between seeing if they're in a code block or not?

I'm guessing if the issue/pr is within a code block, this new event won't get triggered?

Reply:
Yes it won't get triggered, as it was said in the issue, not to trigger this event if it in a codeblock.

Reply:
We can do something similar to 

https://github.com/python-discord/bot/blob/fa02180246e6cf1245d5705b36a2056ed449ef6b/bot/exts/filters/filtering.py#L29-L33

https://github.com/python-discord/bot/blob/fa02180246e6cf1245d5705b36a2056ed449ef6b/bot/exts/filters/filtering.py#L545

Reply:
I think we can combine everything into something like
```
"(```\n?([\s\S]*)?\n?```)|(`([\s\S]*)?`)"
```

Reply:
Alright, will use the one from bot/filters.